# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1375959789 -3600
# Node ID 506035822685f9e9343cbfdb447cdc6b83afe342
# Parent  2bb1e57400cb417dab69cf14a74904e3abe2eb72
rombios/keyboard: Don't needlessly poll the status register

Repeated polling of the status register is not going to change its value, so
don't needlessly take 8192 traps to Qemu when 1 will do.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r 2bb1e57400cb -r 506035822685 tools/firmware/rombios/rombios.c
--- a/tools/firmware/rombios/rombios.c
+++ b/tools/firmware/rombios/rombios.c
@@ -1805,12 +1805,12 @@ keyboard_init()
     while ( (inb(0x64) & 0x02) && (--max>0)) outb(0x80, 0x00);
 
     /* flush incoming keys */
-    max=0x2000;
+    max=2;
     while (--max > 0) {
         outb(0x80, 0x00);
         if (inb(0x64) & 0x01) {
             inb(0x60);
-            max = 0x2000;
+            max = 2;
             }
         }
 
